intercepting-filter/src/main/App.java

```diff
@@ -0,0 +1,245 @@
+		if(request[4].equals("")){
+			return null;
+		}else return request[4];
+	}
+}
+
+class Target extends JFrame{
+	JTable jt;
+	JScrollPane jsp;
+	DefaultTableModel dtm;
+	JButton del;
+	public Target(){
+		super("Order System");
+		setDefaultCloseOperation(EXIT_ON_CLOSE);
+		setSize(640,480);
+		dtm = new DefaultTableModel(new Object[]{"Name", "Contact Number", "Address", "Deposit Number", "Order"},0);
+		jt = new JTable(dtm);
+		del = new JButton("Delete");
+		setup();
+	}
+	private void setup(){
+		setLayout(new BorderLayout());
+		JPanel bot = new JPanel();
+		add(jt.getTableHeader(), BorderLayout.NORTH);
+		bot.setLayout(new BorderLayout());
+		bot.add(del, BorderLayout.EAST);
+		add(bot, BorderLayout.SOUTH);
+		jsp = new JScrollPane(jt);
+		jsp.setPreferredSize(new Dimension(500,250));
+		add(jsp, BorderLayout.CENTER);
+
+		del.addActionListener(new DListener());
+
+		JRootPane rootPane = SwingUtilities.getRootPane(del); 
+		rootPane.setDefaultButton(del);
+		setVisible(true);
+	}
+	public void execute(String[] request){
+		//System.out.println(request[4]);
+		dtm.addRow(new Object[]{request[0],request[1],request[2],request[3],request[4]});
+	}
+
+	class DListener implements ActionListener{
+		@Override
+		public void actionPerformed(ActionEvent e){
+			int temp = jt.getSelectedRow();
+			if(temp == -1) return;
+			int temp2 = jt.getSelectedRowCount();
+			for(int i = 0; i < temp2; i++){
+				dtm.removeRow(temp);
+			}
+		}
+	}
+}
+
+class FilterChain{
+	private ArrayList<Filter> filters = new ArrayList<Filter>();
+	private Target target;
+
+	public void addFilter(Filter filter){
+		filters.add(filter);
+	}
+
+	public String execute(String request){
+		String tempout[] = new String[filters.size()];
+
+		String tempin[] = request.split("&");
+		int i = 0;
+		try{
+			for(Filter filter:filters){
+				tempout[i] = null;
+				tempout[i++] = filter.execute(tempin);
+				//System.out.println(tempout[i]);
+			}
+		}catch(Exception e){
+			return "NOT ENOUGHT INPUT";
+		}
+		
+		if(tempout[4] == null){
+			return "INVALID ORDER!";
+		}else if(tempout[3] == null){
+			return "INVALID DEPOSIT NUMBER!";
+		}else if(tempout[2] == null){
+			return "INVALID ADRDESS!";
+		}else if(tempout[1] == null){
+			return "INVALID Contact Number!";
+		}else if(tempout[0] == null){
+			return "INVALID Name!";
+		}else{
+			target.execute(tempout);
+			return "RUNNING...";
+		}
+	}
+
+	public void setTarget(Target target){
+		this.target = target;
+	}
+}
+
+class FilterManager{
+	FilterChain filterChain;
+
+	public FilterManager(Target target){
+		filterChain = new FilterChain();
+		filterChain.setTarget(target);
+	}
+	public void setFilter(Filter filter){
+		filterChain.addFilter(filter);
+	}
+	public String filterRequest(String request){
+		return filterChain.execute(request);
+	}
+}
+
+class Client extends JFrame{
+	FilterManager filterManager;
+	JLabel jl;
+	JTextField[] jtfarr;
+	JTextArea[] jtaarr;
+	JButton[] buttarr;
```

Comment:
Sorry, but making array of buttons instead of two separate "Clear" and "Process" makes my cry...)
And as we into variable names I see improvements like:
- removing inconsistent "j" prefix (TextField has one, but Button don't)
- removing "arr" suffix and replacing with plural form